internal/vds/vds.cpp

```diff
@@ -551,7 +552,12 @@ struct vdsbuffer fetch_fence(
         const float x = *(coordinates++);
         const float y = *(coordinates++);
 
-        const auto coordinate = transform_coordinate(x, y);
+        auto coordinate = transform_coordinate(x, y);
+
+        if (interpolation_method == NEAREST) {
```

Comment:
This is one of the few cases where I would appreciate a inline comment explaining why it's necessary to do this rounding